simpleast-core/src/test/java/com/discord/simpleast/code/JavaScriptRuleTest.kt

```diff
@@ -0,0 +1,225 @@
+
+  private class TestState
+
+  private lateinit var parser: Parser<TestRenderContext, Node<TestRenderContext>, TestState>
+  private lateinit var treeMatcher: TreeMatcher
+
+  @Before
+  fun setup() {
+    val codeStyleProviders = CodeStyleProviders<TestRenderContext>()
+    parser = Parser()
+    parser
+        .addRule(CodeRules.createCodeRule(
+            codeStyleProviders.defaultStyleProvider,
+            CodeRules.createCodeLanguageMap(codeStyleProviders))
+        )
+        .addRules(SimpleMarkdownRules.createSimpleMarkdownRules())
+    treeMatcher = TreeMatcher()
+    treeMatcher.registerDefaultMatchers()
+  }
+
+  @Test
+  fun comments() {
+    val ast = parser.parse("""
+      ```js
+      /** Multiline
+          Comment
+      */
+      foo.bar(); // Inlined
+      // Line comment
+      
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        """
+          /** Multiline
+              Comment
+          */
+        """.trimIndent(),
+        "// Inlined",
+        "// Line comment")
+  }
+
+  @Test
+  fun strings() {
+    val ast = parser.parse("""
+      ```js
+      const x = 'Hello';
+      const y = "world";
+      console.log(`Hi`);
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "'Hello'",
+        "\"world\"",
+        "`Hi`")
+  }
+  @Test
+  fun stringsMultiline() {
+    val ast = parser.parse("""
+      ```js
+      const text = `
+      hello
+      world
+      `;
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "`",
+        """
+          hello
+          world
+        """.trimIndent(),
+        "`"
+    )
+  }
+
+  @Test
+  fun functions() {
+    val ast = parser.parse("""
+      ```js
+      function test(T) {
+        // Implementation
+      }
+      const fn = function() {};
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<JavaScript.FunctionNode<*>>(
+      """function test(T) {
+           // Implementation
+         }""".trimIndent(),
+      "function() {}")
+  }
+
+  @Test
+  fun commentedFunction() {
+    val ast = parser.parse("""
+      ```js
+      /*
+        function test(T) {
+          throw new Error();
+        }
+      */
+      // function O() {}
+      console.log(x /* test var */);
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        """
+        /*
+          function test(T) {
+            throw new Error();
+          }
+        */
+        """.trimIndent(),
+        "// function O() {}",
+        "/* test var */")
+  }
+
+  @Test
+  fun keywords() {
+    val ast = parser.parse("""
+      ```js
+      const x = 'Test';
+      while (true) {}
+      for (;;) {}
+      if (false) {}
+      class Test {}
+      try {
+        // Nothing
+      } catch (err) {
+        console.error(err);
```

Comment:
```suggestion
        throw
```
Convey's the same idea, just avoids using `console`